### DISCUSSION TITLE
Shopware 6.5.3.0

### DISCUSSION THREAD

**Discussion by shyim** [shopware/platform#3201] [Announcements]
Hey 👋 ,

We released Shopware 6.5.3.0 today. Here are some highlights:

### Improvements to HTML Sanitizer

We got a lot of feedback on the HTML Sanitizer, which was introduced with Shopware 6.5. Based on this feedback, we adjusted the Sanitizer to allow more elements and attributes. Furthermore with this release it is also possible to turn off this feature completely:

```yaml
# config/packages/z-shopware.yml
shopware:
    html_sanitizer:
        enabled: false

```

For more information on this, [refer to the docs](https://developer.shopware.com/docs/guides/security/html-sanitizer).

### Improved logging

We have excluded a bunch of Exception which are triggered by "wrong" user interactions, for example the `PluginNotInstalledException`.

## UUIDv7

With this release we changed the default UUID format to UUIDv7, to improve the insert performance of the Database. In the local benchmarks this change brings a reduction of 138s to 121s when creating 10000 demo data products.

See #3086 for more information.

## Improved Admin Worker

The Admin Worker is a background process in your Administration which mimics an CLI worker to handle tasks asynchronously.
This worker has been run for each administration user, for each open administration tab and therefore consumed too many resources, when the queue was full. With this release, only one administration worker runs at the same time.
**[In general, it is recommended to all systems, to switch to the CLI worker](https://developer.shopware.com/docs/guides/plugins/plugins/framework/message-queue/add-message-handler#the-cli-worker)**.

## Application Wide Persistent KeyValueStorage

Since the beginning of Shopware 6, we save installation-specific configuration like the cache-id in the `app_config` table. Now we added finally an abstraction around this: `\Shopware\Core\Framework\Adapter\Storage\AbstractKeyValueStorage`, such that this storage can be replaced.

```php
abstract class AbstractKeyValueStorage
{
    abstract public function has(string $key): bool;

    abstract public function get(string $key, mixed $default = null): mixed;

    abstract public function set(string $key, mixed $value): void;

    abstract public function remove(string $key): void;
}
```

### Cookie Consent for YouTube Videos

We implemented YouTube videos into the Cookie Consent, so the Users has to accept the consent before the YouTube Video will be loaded:

![video-elment](https://github.com/shopware/platform/assets/6224096/788fca88-53ad-4fd4-bd59-2bb24cae1919)

### And many more things

- Improved CleanupCartTaskHandler performance #3069
- Fixed Custom Fields indexing in Elasticsearch
- Fixed Elasticsearch Date parsing, #3095
- Fix serialization of mails with attachments into the Message Queue
- Import Export allows now importing media which are url encoded #3062

For the complete list of commits, refer to the actual [Release](https://github.com/shopware/platform/releases/tag/v6.5.3.0)

**Comment by rattananen** [comment]
Is UUID v7 compactible with UUID v4?
What's happen when exist shop that use v4 update to v7? 
Do we need to migrate  exist v4 to v7? 
Is there chance to duplicate when use together with UUID v4?
How we can generate v7 format in javascript? 

I also research about this It's truly better performance but I think best way is keep to v4. 
for feature that we can easily generate UUID  everywhere without duplicate . 
and v4 is widely use Shopware could migrate UUID from external system without worry to duplicate.

in javascript we can generate v4 without any library by `Crypto.randomUUID() ` (just remove `-` then it will be same format with v4 that Shopware use)

The Shopware is also generate UUID v4 in javascript but in the commit #3086 has no update for UUID in javascript.
This is very risky commit.

and Symfony also has UUID component why we need to write and maintain by own https://symfony.com/doc/current/components/uid.html .

**Reply by Christian-Rades** [reply]
To answer your question directly, UUIDv7 is fully compatible with UUIDv4. The only difference between the two is the version bits in the middle of the ID types. Without comparing the version bits, every UUIDv7 is a valid UUIDv4 and vice versa.

In terms of migrating, there is no need to migrate an existing v4 UUID to v7 as Shopware does not check the version byte anywhere. This won't change in the foreseeable future as a lot of data imports rely on using a hash of external keys to generate UUIDs for Shopware.

To put this issue into more relatable terms:
- A collision between v7 and v4 is logically impossible as both contain a version marker
    - A collision between two v4 IDs has a probability of 2 to the power of 122
        - Assuming there are 10^20 grains of sand on the earth there are 53169119831396634916152282411213783 IDs per grain of sand so on the order of 10^34
        - v7 has “only” 62 pseudo-random bits. That means roughly there are 10^18 IDs for each  second since 01.01.1970 as there is a time-specific prefix (that also can hold additional 12 bits of sub-second precision time)

We have chosen to implement our own version of the generator to reduce the amount of abstraction involved in generating such a fundamental value in our system. We also expect little maintenance effort to come from this code as the specification is quite solid and the system is very permissive with regards to ID layouts.

It’s true that there was an oversight concerning the js UUID method, but it is fixable in the future.

**Reply by rattananen** [reply]
This is clearly answer. Thank you.

I had found structure for UUID.
format for each version in hex will be `xxxxxxxx-xxxx-Vxxx-yxxx-xxxxxxxxxxxx` .  `V` is version.
so v4 will be  `xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx`
and v7  ` xxxxxxxx-xxxx-7xxx-yxxx-xxxxxxxxxxxx`
This make, they are impossible to duplicate when we use both in same field.
I just shorten your explanation.

Actually, we can check version of UUID that given follow this structure.
